cocos/scripting/js-bindings/script/studio/parsers/timelineParser-2.x.js

```diff
@@ -1390,28 +1416,49 @@
-                case "USER2":
-                    node.setCameraFlag(cc.CameraFlag.USER2); break;
-                case "USER3":
-                    node.setCameraFlag(cc.CameraFlag.USER3); break;
-                case "USER4":
-                    node.setCameraFlag(cc.CameraFlag.USER4); break;
-                case "USER5":
-                    node.setCameraFlag(cc.CameraFlag.USER5); break;
-                case "USER6":
-                    node.setCameraFlag(cc.CameraFlag.USER6); break;
-                case "USER7":
-                    node.setCameraFlag(cc.CameraFlag.USER7); break;
-                case "USER8":
-                    node.setCameraFlag(cc.CameraFlag.USER8); break;
-                case "DEFAULT":
-                    node.setCameraFlag(cc.CameraFlag.DEFAULT); break;
-                default:
-                    node.setCameraFlag(cc.CameraFlag.USER1);
+            var cameraFlagData = json["CameraFlagData"];
+            var cameraFlag = cc.CameraFlag.USER1;
+            if(undefined === cameraFlagData || isNaN(cameraFlagData) || 0 === cameraFlagData)
+            {
+                switch(camMode){
+                    case "USER1":
+                        cameraFlag = cc.CameraFlag.USER1; break;
+                    case "USER2":
+                        cameraFlag = cc.CameraFlag.USER2; break;
+                    case "USER3":
+                        cameraFlag = cc.CameraFlag.USER3; break;
+                    case "USER4":
+                        cameraFlag = cc.CameraFlag.USER4; break;
+                    case "USER5":
+                        cameraFlag = cc.CameraFlag.USER5; break;
+                    case "USER6":
+                        cameraFlag = cc.CameraFlag.USER6; break;
+                    case "USER7":
+                        cameraFlag = cc.CameraFlag.USER7; break;
+                    case "USER8":
+                        cameraFlag = cc.CameraFlag.USER8; break;
+                    case "DEFAULT":
+                        cameraFlag = cc.CameraFlag.DEFAULT; break;
+                }
+            } else {
+                cameraFlag = cameraFlagData;
             }
+            node.setCameraFlag(cameraFlag);
+        }
+
+        if(json.hasOwnProperty("SkyBoxEnabled") && true == json["SkyBoxEnabled"])
+        {
+            var leftFileData = getSkyboxRes(json, "LeftImage");
+            var rightFileData = getSkyboxRes(json, "RightImage");
+            var upFileData = getSkyboxRes(json, "UpImage");
+            var downFileData = getSkyboxRes(json, "DownImage");
+            var forwardFileData = getSkyboxRes(json, "ForwardImage");
+            var backFileData = getSkyboxRes(json, "BackImage");
+
+            var skyBox = new jsb.Skybox();
+            skyBox.init(leftFileData,rightFileData,upFileData,downFileData,forwardFileData,backFileData);
+            skyBox.setCameraMask(cameraFlag);
+            node.addChild(skyBox, 0, "_innerSkyBox");
```

Comment:
OK, I'll submit new PR to this branch first, and I have just check v3.7.1 branch, seems skybox parser haven't been merged to it, should I submit a full skybox reader pr to v3.7.1?